gogen/goelements.go

```diff
@@ -642,16 +642,39 @@ func (s *GoLangMapper) yangDefaultValueToGo(value string, args resolveTypeArgs,
 		if err != nil {
 			return "", yang.Ynone, err
 		}
+
+		// Check if the value is an empty string.
+		if len(value) == 0 {
+			return "", yang.Ynone, fmt.Errorf("default value conversion: empty value")
+		}
+
+		// When the default value of int/uint is not base-10, check if the base should be supported according to
+		// https://datatracker.ietf.org/doc/html/rfc7950#section-9.2.1
+		//
+		// When the first character in the value is `0`, the value could be hexadecimal, octal or binary. While hexadecimal
+		// and octal values are supported (with an optional sign prefix), binary is not allowed by the RFC. The format support
+		// of `strconv.ParseInt/ParseUint` functions should be further constrained.
+		if value[0] == '0' {
+			if len(value) > 1 && (value[1] == 'b' || value[1] == 'B') {
+				return "", yang.Ynone, fmt.Errorf("default value conversion: base 2 value `%s` is not allowed", value)
+			}
+		}
+
+		// Underscores are not allowed in the value, although `strconv` allows them in certain formats.
+		if strings.Contains(value, "_") {
+			return "", yang.Ynone, fmt.Errorf("default value conversion: `_` is not allowed in the value %s", value)
+		}
+
```

Comment:
Instead of checking for negative cases, it might be cleaner to check for positive cases and then use ParseInt's error handling to deal with all other unexpected cases. So I'm thinking replacing this highlighted block with

```go
base := 10
switch {
case strings.HasPrefix(value, "0"):
  base = 8
case strings.HasPrefix(value, "0x"):
  base = 16
  value = strings.TrimLeft(value, "0x")
}

Reply:
The RFC allows use of an optional `+/-` sign prefix for octal and hexadecimal values, so it will need a few more lines to handle all cases... In fact the previous commit needs to handle the binary negative case with a possible sign prefix too.  Personally I'd prefer adding the sign handling to the previous commit's negative case handling. But let me know your thoughts. Thanks.

Reply:
Thanks Jay, and sorry for the delay in review. I'm open to keeping the current PR but I'm wondering how difficult it is to check the positive cases?

```go
if strings.HasPrefix(value, "+") {
  value = value[1:]
}

base := 10
switch {
case strings.HasPrefix(value, "0"), strings.HasPrefix(value, "-0"):
  base = 8
case strings.HasPrefix(value, "0x"), ...:
  base = 16
  value = strings.TrimLeft(value, "0x")
}
```

This looks shorter than the negative case handling and is more understandable. Are there other cases which I've missed here?

Reply:
Thanks! This should work similarly : ) But I think it still needs a couple more lines to return an error when seeing values with underscores.

Reply:
The binary format `0b` needs to be checked before the switch statement too. Otherwise, the first switch case will let `0b` pass through and give it base 8.

Reply:
I see, even the current code is not handling the underscores. I think this looks fine to me, thanks!